bedrock/base/templates/macros.html

```diff
@@ -236,3 +236,84 @@ <h2 class="thank-you hidden">{{ _('Your download link was sent.') }}</h2>
     </footer>
   </section>
 {%- endmacro %}
+
+{% macro fxa_create_account_button(button_location, id_prefix, campaign, entrypoint, content = '' ) %}
+  {# - creates the "create account" button which links to accounts.firefox.com
+       and fallback states including:
+       - already logged in, not firefox, legacy firefox, Android Firefox, iOS Firefox
+       - use with media/js/base/mozilla-fxa.js and media/css/pebbles/components/_fxa.scss
+     - button_location and id_prefix are required
+       - button_location is for `data-download-location` which is for analytics
+          - examples: 'primary cta', 'footer cta'
+       - id_prefix is combined with some sufixes to create unique IDs for all buttons and links
+         - example: 'features-header'
+     - campaign is for utm_campaign on the Fxa and fallback links
+     - entrypoint is for the fxa team, it indicates where the user came from
+       - the prefix `mozilla.org:` will be appended to it
+       - example: 'fxa.features'
+       - when in doubt ask the fxa team to recommend a value
+     - content is for utm_content param
+       - content will default to the path to the page, minus the locale
+  #}
+  {# Parameters for accounts.firefox.com link #}
+  {% set service = 'sync' %}
+  {% set context = 'fx_desktop_v3' %}
+  {% set entrypoint = 'mozilla.org:' + entrypoint %}
+  {% if content %}
+    {% set utm_content = content %}
+  {% else %}
+    {% set utm_content = request.path_info.replace('/', '%2F') %}
+  {% endif %}
+
+  {# Create accounts.firefox.com link #}
+  {% set accounts_url = ['https://accounts.firefox.com/signup?service=', service,
+                          '&context=', context,
+                          '&entrypoint=', entrypoint,
+                          '&utm_content=', utm_content,
+                          '&utm_campaign=', campaign
+                        ]|join %}
+
+  {# Create ids for buttons and links #}
+  {% set download_id = '{}-download'.format(id_prefix) %}
+  {% set account_id = '{}-account'.format(id_prefix) %}
+  {% set android_id = '{}-android'.format(id_prefix) %}
+  {% set ios_id = '{}-ios'.format(id_prefix) %}
+  {% set update_id = '{}-update'.format(id_prefix) %}
+  {% set link_id = '{}-link'.format(id_prefix) %}
+
+  {# campaign for iOS app store link #}
+  {% set app_store_campaign = campaign + '-appstore-button' %}
+
+  <div class="show-fxa-not-fx show-fxa-default">
+    {{ download_firefox(dom_id=download_id, download_location=button_location) }}
+  </div>
+  <div class="show-fxa-supported-signed-out">
+    <a href="{{ accounts_url }}" class="button button-hollow button-light" data-button-name="Create Account" data-link-type="button" data-cta-position="{{ button_location }}" id="{{ account_id }}">
```

Comment:
IIUC, this new macro is creating an anchor with an URL, instead of the relying on the `Mozilla.UITour.showFirefoxAccounts` call?

Would you please create an MozillaOnline equivalent (that is, "https://accounts.firefox.com.cn/...") and set it as "data-mozillaonline-link" attribute on this anchor? Please see #4066 & #4067 for more backgrounds. Thanks!

Reply:
@shane-tomlinson or @jpetto Is it better to use the `Mozilla.UITour.showFirefoxAccounts` than link directly? I can implement that for the new macro if it's better. I was avoiding it because the behaviour has changed from [what we have documented](http://bedrock.readthedocs.io/en/latest/uitour.html#showfirefoxaccounts-extraurlcampaignparams) and because we can't set the entry point.

It's also easy to add the `data-mozillaonline-link` if the link is fine.

Reply:
I'm have to defer to @jpetto on the best way to go about things in bedrock.

Reply:
IMO we shouldn't use UITour unless we have to. +1 to using a direct link.

Reply:
👍